### PR TITLE
has_trait function

### PR DESCRIPTION
**Pull request by SylvainCorlay**
As per discussion with @takluyver .

### CODE REVIEW

**Comment by minrk** [comment]
:+1:, but can you add a simple exercise test, for good measure?

**Comment by SylvainCorlay** [comment]
alright :)

File: traitlets/traitlets.py [hunk 1]
```diff
@@ -801,6 +800,12 @@ def class_own_traits(cls, **metadata):
         return {n: t for (n, t) in cls.class_traits(**metadata).items()
                 if getattr(sup, n, None) is not t}
 
+    def has_trait(self, name):
+        """Returns True if the specified name is the name of a trait
+        attribute."""
+        return (hasattr(self.__class__, name) and
+                isinstance(getattr(self.__class__, name), TraitType))
```
Comment (takluyver) [inline traitlets/traitlets.py]: getattr takes an optional default parameter, so we should be able to save the separate `hasattr` check and just do `isinstance(getattr(self.__class__, name, None), TraitType)`.

File: traitlets/traitlets.py [hunk 2]
```diff
@@ -801,6 +800,12 @@ def class_own_traits(cls, **metadata):
         return {n: t for (n, t) in cls.class_traits(**metadata).items()
                 if getattr(sup, n, None) is not t}
 
+    def has_trait(self, name):
+        """Returns True if the specified name is the name of a trait
+        attribute."""
```
Comment (takluyver) [inline traitlets/traitlets.py]: The summary in a docstring should generally be kept to one line. Something like "Returns True if the object has a trait with the specified name."

**Comment by SylvainCorlay** [comment]
done, squashed in a single commit.

**Comment by takluyver** [comment]
Can you also add the method to the HasTraits description in the docs (`using_traitlets.rst`)? Sorry I didn't think of that before?

**Comment by SylvainCorlay** [comment]
Done

**Comment by takluyver** [comment]
Thanks

**Pull request merged**